src/test/TestTransactionFrame.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+// Copyright 2022 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/FeeBumpTransactionFrame.h"
+#include "transactions/TransactionFrame.h"
+
+namespace stellar
+{
+
+class TestTransactionFrame : public TransactionFrame
```

Comment:
This class can be reworked to avoid more copy/paste down the line: you can just add a member `std::shared_ptr<TransactionFrameBase> mTx` and derive from `TransactionFrameBase` -> you can implement a proxy pattern and update `mResult` as needed.

Reply:
update: I am not even sure you should derive from `TransactionFrameBase`.
* test code needs the version of the methods that don't have a "result"
* the whole `syncResult` thing is misleading and does not catch all bugs if people are mixing using a local variable and using this class' member, so there is no real point in trying to keep those things in sync
* for the few places where you need a `TransactionFrameBase` , you can just get a `const` reference to `mTx`.